PukingPredator/Assets/Scripts/Physics/CollapseIfHeavyPlayer.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CollapseIfHeavyPlayer : MonoBehaviour
+{
+    // Put on tiles with PhysicsBehavior and adjust the massThreshold
+
+    private PhysicsBehaviour pb;
+
+    /// <summary>
+    /// If player is greater than or equal to this mass when they collide with the structure, collapse the structure.
+    /// </summary>
+    [SerializeField]
+    private float massThreshold = 5f;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        pb = GetComponent<PhysicsBehaviour>();
+    }
+
+    private void OnCollisionEnter(Collision collision)
+    {
+        if (collision.collider.CompareTag("Player"))
```

Comment:
we have something like "GameTag.player" that should be used in place of "Player"